### PR TITLE
ref(ui): Switch from clipboard to copy-text-to-clipboard

### PR DESCRIPTION
**Pull request by scttcper**
It's smaller, doesn't need ReactDOM to copy. For text copying it is using the same api as the previous library but does not provide a way to check that the api used is available. Its one line to port that over `document.queryCommandSupported('copy');`

[copy-text-to-clipboard](https://github.com/sindresorhus/copy-text-to-clipboard) is by sindresorhus, we already use other libraries of theirs like query-string

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/clipboard.tsx [hunk 1]
```diff
@@ -1,91 +1,64 @@
...
+}
 
-  componentWillUnmount() {
-    if (this.clipboard) {
-      this.clipboard.destroy();
+function Clipboard({
+  children,
+  value,
+  successMessage = 'Copied to clipboard',
+  errorMessage = 'Error copying to clipboard',
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/clipboard.tsx:29]: As we your editing this file, can we maybe update these default values?

```suggestion
  successMessage = t('Copied to clipboard'),
  errorMessage = t('Error copying to clipboard'),
```

File: src/sentry/static/sentry/app/components/clipboard.tsx [hunk 2]
```diff
@@ -1,91 +1,64 @@
...
+    if (success) {
+      if (!hideMessages) {
+        addSuccessMessage(successMessage);
+      }
+      onSuccess?.();
+    } else {
+      if (!hideMessages) {
+        addErrorMessage(errorMessage);
+      }
+      onError?.();
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/clipboard.tsx:47]: can we maybe remove the `else` ?

```suggestion
    if (success) {
      if (!hideMessages) {
        addSuccessMessage(successMessage);
      }
      onSuccess?.();
      return;
    } 
    
      if (!hideMessages) {
        addErrorMessage(errorMessage);
      }
      onError?.();
```

File: src/sentry/static/sentry/app/components/clipboard.tsx [hunk 3]
```diff
@@ -1,91 +1,64 @@
...
 
-    return React.cloneElement(children, {
-      ref: this.handleMount,
-    });
+  if (!React.isValidElement(children)) {
+    return null;
   }
+
+  return React.cloneElement(children, {
+    onClick: () => handleClick(),
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/clipboard.tsx:60]: can we just assign the `handleClick` here?

```suggestion
    onClick: handleClick,
```

**Review by billyvg — APPROVED** [review]
good stuff

**Pull request merged**